### PR TITLE
Add missing "await" prefix or "Async" suffix in "async" methods

### PR DESCRIPTION
**Pull request by marco-carvalho**

### CODE REVIEW

**Review by IsaacMarovitz — APPROVED** [review]

File: src/Ryujinx.Common/Utilities/StreamUtils.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public static IMemoryOwner<byte> StreamToRentedMemory(Stream input)
 
         public static async Task<byte[]> StreamToBytesAsync(Stream input, CancellationToken cancellationToken = default)
         {
-            using MemoryStream stream = MemoryStreamManager.Shared.GetStream();
+            await using MemoryStream stream = MemoryStreamManager.Shared.GetStream();
```
Comment (jhorv) [inline src/Ryujinx.Common/Utilities/StreamUtils.cs:62]: There's no point in async-disposing of a MemoryStream, it just adds overhead around calling Dispose().

File: src/Ryujinx/UI/ViewModels/AmiiboWindowViewModel.cs [hunk 2]
```diff
@@ -479,7 +479,7 @@ private async Task UpdateAmiiboPreview(string imageUrl)
             if (response.IsSuccessStatusCode)
             {
                 byte[] amiiboPreviewBytes = await response.Content.ReadAsByteArrayAsync();
-                using MemoryStream memoryStream = new(amiiboPreviewBytes);
+                await using MemoryStream memoryStream = new(amiiboPreviewBytes);
```
Comment (jhorv) [inline src/Ryujinx/UI/ViewModels/AmiiboWindowViewModel.cs:482]: As noted elsewhere, async-disposing a `MemoryStream` only adds overhead.

File: src/Ryujinx.Common/Utilities/EmbeddedResources.cs [hunk 3]
```diff
@@ -97,7 +97,7 @@ public static string ReadAllText(Assembly assembly, string filename)
 
         public async static Task<string> ReadAllTextAsync(Assembly assembly, string filename)
         {
-            using var stream = GetStream(assembly, filename);
+            await using var stream = GetStream(assembly, filename);
```
Comment (jhorv) [inline src/Ryujinx.Common/Utilities/EmbeddedResources.cs:100]: These `Stream`s returned by `GetStream()` -> `Assembly.GetManifestResourceStream()` are based on `UnmanagedMemoryStream`, and disposing asynchronously only adds overhead.

File: src/Ryujinx.Common/Utilities/StreamUtils.cs (hunk 1 above)
Comment (marco-carvalho) [reply src/Ryujinx.Common/Utilities/StreamUtils.cs]: done

File: src/Ryujinx/UI/ViewModels/AmiiboWindowViewModel.cs (hunk 2 above)
Comment (marco-carvalho) [reply src/Ryujinx/UI/ViewModels/AmiiboWindowViewModel.cs]: done

File: src/Ryujinx.Common/Utilities/EmbeddedResources.cs (hunk 3 above)
Comment (marco-carvalho) [reply src/Ryujinx.Common/Utilities/EmbeddedResources.cs]: done

**Pull request closed without merging**